app/endpoints/generated_endpoints.py

```diff
@@ -64,87 +55,61 @@ def create_endpoint_for_form(
-        def endpoint(
-            db_client: Annotated[entitysdk.client.Client, Depends(get_client)],
-            form: model,
-        ) -> str:
-            L.info("generate_grid_scan")
-            L.info(db_client)
-
-            campaign = None
-            try:
-                with tempfile.TemporaryDirectory() as tdir:
-                    grid_scan = GridScan(
-                        form=form,
-                        # TODO: output_root=settings.OUTPUT_DIR / "fastapi_test" / model_name
-                        #        / "grid_scan", => ERA001 Found commented-out code
-                        output_root=tdir,
-                        coordinate_directory_option="ZERO_INDEX",
-                    )
-                    campaign = grid_scan.execute(
-                        processing_method=processing_method,
-                        data_postprocessing_method=data_postprocessing_method,
-                        db_client=db_client,
-                    )
-
-            except Exception as e:
-                error_msg = str(e)
-
-                if len(e.args) == 1:
-                    error_msg = str(e.args[0])
-                elif len(e.args) > 1:
-                    error_msg = str(e.args)
-
-                L.error(error_msg)
-
-                raise HTTPException(status_code=500, detail=error_msg) from e
-
-            else:
-                L.info("Grid scan generated successfully")
-                if campaign is not None:
-                    return str(campaign.id)
-
-                L.info("No campaign generated")
-                return ""
+    if isinstance(return_class, str):
+        return
+    endpoint_name_with_slash = "/" + model_name + "-" + processing_method + "-grid"
+    if data_postprocessing_method:
+        endpoint_name_with_slash = endpoint_name_with_slash + "-" + data_postprocessing_method
+
+    @router.post(endpoint_name_with_slash, summary=model.name, description=model.description)
+    def endpoint(
+        db_client: Annotated[entitysdk.client.Client, Depends(get_client)],
+        form: model,
+    ) -> str:
+        campaign = None
+        try:
+            with tempfile.TemporaryDirectory() as tdir:
+                grid_scan = GridScan(
+                    form=form,
+                    output_root=tdir,
+                    coordinate_directory_option="ZERO_INDEX",
+                )
+                campaign = grid_scan.execute(
+                    processing_method=processing_method,
+                    data_postprocessing_method=data_postprocessing_method,
+                    db_client=db_client,
+                )
+        except Exception as e:
+            error_msg = str(e)
+            if len(e.args) == 1:
+                error_msg = str(e.args[0])
+            elif len(e.args) > 1:
+                error_msg = str(e.args)
+            raise HTTPException(status_code=500, detail=error_msg) from e
```

Comment:
I don't know why all of this is in this code review.

Also the status_code = 500 looks sus, and a post method just returning a str is unusual.

Reply:
This is the version from main. I don't know the rationale behind these design choices.

Reply:
It is rebased.